### PR TITLE
Gh 74 fix released lock

### LINKED ISSUE #74: v1.0.0 accidentally releases lock on subprocesses

### PR DESCRIPTION
**Pull request by kentwills**
https://bpaste.net/show/0f908c9eb4e5
closes #74

### CODE REVIEW

**Comment by bukzor** [comment]
"All checks have failed"

File: tests/examples/.gitignore [hunk 1]
```diff
@@ -0,0 +1 @@
+notification-fd
```
Comment (bukzor) [inline tests/examples/.gitignore]: i'm not sure there should be a .gitignore here

File: tests/examples/orphan-subprocess/playground/slow-startup/service [hunk 2]
```diff
@@ -0,0 +1,4 @@
+#!/bin/bash
+sleep 2
+touch readyfile
+exec sleep infinity
```
Comment (bukzor) [inline tests/examples/orphan-subprocess/playground/slow-startup/service]: there's no "orphan-subprocess" here.

Maybe you want `sleep 9999 &` as the first lien.

**Comment by kentwills** [comment]
@bukzor thanks for those wise words of wisdom

File: tests/examples/orphan-subprocess/playground/slow-startup/service (hunk 2 above)
Comment (kentwills) [inline tests/examples/orphan-subprocess/playground/slow-startup/service]: did this not reproducing, thought I had centered on this one

File: tests/examples/orphan-subprocess/playground/slow-startup/service [hunk 3]
```diff
@@ -0,0 +1,6 @@
+#!/bin/bash
+cd ../../
+sleep 2
+touch readyfile
```
Comment (bukzor) [inline tests/examples/orphan-subprocess/playground/slow-startup/service]: This seems more clean:

```
(cd /; sleep 987654) &  # an orphan subprocess, that doesn't hold any reference to $PWD
```

**Pull request merged**